java/src/main/native/ai_onnxruntime_OrtSession.c

```diff
@@ -394,21 +405,26 @@ JNIEXPORT jobjectArray JNICALL Java_ai_onnxruntime_OrtSession_run(JNIEnv* jniEnv
     goto cleanup_output_values;
   }
 
-  // Construct the output array of ONNXValues
-  jclass onnxValueClass = (*jniEnv)->FindClass(jniEnv, ORTJNI_OnnxValueClassName);
-  outputArray = (*jniEnv)->NewObjectArray(jniEnv, safecast_int64_to_jsize(numOutputs), onnxValueClass, NULL);
+  // Create the output boolean array denoting if ORT owns the memory for each output.
+  // Java boolean arrays are initialized to false.
+  outputArray = (*jniEnv)->NewBooleanArray(jniEnv, safecast_int64_to_jsize(numOutputs));
+  jboolean* boolArr = (*jniEnv)->GetBooleanArrayElements(jniEnv, outputArray, NULL);
```

Comment:
>jboolean* boolArr [](http://example.com/codeflow?start=2&length=17)

The purpose of this array is not clear.

Reply:
It's a reference to the boolean array on the Java side which we use to track if this is a user supplied output or one ORT created in the run call. It makes the accounting a little easier back on the Java side as we write to the array of OnnxValues here on the C side so we needed either to copy it on the Java side and compare against null or allocate some other way of tracking it (which is what this is).